test/system/src/test/java/io/pravega/test/system/KeyValueStoreTest.java

```diff
@@ -0,0 +1,744 @@
+        List<URI> ctlURIs = controllerInstance.getServiceDetails();
+        final List<String> uris = ctlURIs.stream().filter(ISGRPC).map(URI::getAuthority).collect(Collectors.toList());
+        controllerURI = URI.create("tcp://" + String.join(",", uris));
+        final ClientConfig clientConfig = Utils.buildClientConfig(controllerURI);
+        final ScheduledExecutorService controllerExecutor= Executors.newScheduledThreadPool(5);
+        this.controller = new ControllerImpl(ControllerImplConfig.builder().clientConfig(clientConfig).build(),controllerExecutor);
+        this.connectionFactory = new ConnectionFactoryImpl(clientConfig);
+        boolean createdScope = this.controller.createScope(SCOPE_NAME).join();
+        assertTrue("The requested Scope must get created", createdScope);
+        log.info("Scope {} created successfully", SCOPE_NAME);
+        this.keyValueTableFactory = new KeyValueTableFactoryImpl(SCOPE_NAME,this.controller,this.connectionFactory);
+        boolean created = this.controller.createKeyValueTable(KVT.getScope(), KVT.getKeyValueTableName(), config).join();
+        Assert.assertTrue("The requested KVT must get created",created);
+        log.info("KVT {} created successfully", KVT_NAME);
+        this.keyValueTable = this.keyValueTableFactory.forKeyValueTable(KVT.getKeyValueTableName(), KEY_SERIALIZER, VALUE_SERIALIZER,
+                KeyValueTableClientConfiguration.builder().build());
+
+    }
+
+    @After
+    public void tearDown() {
+        this.connectionFactory.close();
+        this.controller.close();
+        this.keyValueTableFactory.close();
+        this.keyValueTable.close();
+    }
+
+    // Test case-42: Insert Same Key into multiple KeyFamilies
+    @Test
+    public void test1SameKeyMutipleKF() throws UnsupportedOperationException {
+        try {
+            String kf1 = "KeyFamily1";
+            String kf2 = "KeyFamily2";
+
+            Integer key = 22;
+            String value1 = "This value belongs to KeyFamily1";
+            String value2 = "This value belongs to KeyFamily2";
+
+            Version version1 = this.keyValueTable.put(kf1, key, value1).join();
+            Version version2 = this.keyValueTable.put(kf2, key, value2).join();
+            log.info("Version1: {}", version1);
+            log.info("Version2: {}", version2);
+
+            TableEntry<Integer, String> tableEntry1 = this.keyValueTable.get(kf1, key).join();
+            TableEntry<Integer, String> tableEntry2 = this.keyValueTable.get(kf2, key).join();
+
+            assertEquals("Keys should be same for both entries",
+                    tableEntry1.getKey().getKey(), tableEntry2.getKey().getKey());
+            assertTrue("Corresponding value for entry1 should be as inserted",
+                    value1.equals(tableEntry1.getValue()));
+            log.info("Corresponding value '{}' for entry1 is inserted '{}'", value1, tableEntry1.getValue());
```

Comment:
What's the point of this log.info? You have already validated that the two values are equal so why log it?

Reply:
Just added to verify the logs if it passes the assert check. Will remove now